src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -263,6 +344,40 @@ internal Location CreateLocation(LocationVersionOne v1Location)
             return location;
         }
 
+        internal Location CreateLocation(AnnotatedCodeLocationVersionOne v1AnnotatedCodeLocation)
+        {
+            Location location = null;
+
+            if (v1AnnotatedCodeLocation != null)
+            {
+                location = new Location
+                {
+                    Annotations = v1AnnotatedCodeLocation.Annotations?.SelectMany(a => a.Locations, (a, pl) => CreateRegion(pl)).ToList(),
```

Comment:
Oh, wow, very nice. I had to look it up to remind myself that v1 `location.annotations` is of type `region[]`.

But there's a bug. The v1 `annotation`  object contained a message, and a set of locations to which the message applies. It was a terrible design for reasons I can go into offline if you like.

The point is, in the new design -- where the `region` object has a `message` property -- that v1 `annotatedCodeLocation.message` needs to go on each of the v2 `region`s you create. As the code stands, the message is lost.

We need to enhance the tests to detect this (by adding a couple of v1 `annotations` to one of the v1 `annotatedCodeLocation`s.

Reply:
Gotcha

Reply:
Done